src/java.base/share/classes/java/lang/StringBuffer.java

```diff
@@ -712,9 +712,7 @@ public synchronized StringBuffer reverse() {
     @IntrinsicCandidate
     public synchronized String toString() {
         if (toStringCache == null) {
-            return toStringCache =
-                    isLatin1() ? StringLatin1.newString(value, 0, count)
-                               : StringUTF16.newString(value, 0, count);
+            return toStringCache = new String(this);
```

Comment:
When you modify a method labeled `IntrinsicCandidate`, you should probably modify the intrinsic as well

Reply:
_StringBuilder_toString and _StringBuffer_toString are intrinsics indeed, but they are only used for method recognition in c2/StringOpts. C2 doesn't expand them. if StringOpts misses, C2 will inline these 2 methods.